### ISSUE TITLE
Create Javadoc gh pages within CI process

### ISSUE THREAD

**Issue by bestmazzo** [freedomotic/freedomotic#133]
As @mcicolella proposed, this could be a good start:
http://blog.progs.be/517/publishing-javadoc-to-github-using-maven

**Comment by Helltux** [comment]
I take this

**Comment by Helltux** [comment]
ok, thanks for the blog article but i think we need something to discuss. 
You can see, i have set in the pom.xml the git remote repo. maybe we find a way to set this over a variable, to the git-user of the fork. Then everyone can make a javadoc an push to his own fork and make then a PR to the master.
Or we set this to the root of the project and only some one they have permission can make an update on the javadoc. What i think is the better way. Because of the first way you have a lot of JavaDoc pages in the www an i think they will
find over search engine

Pom.xml
https://github.com/Helltux/freedomotic/blob/ccf53c36f2725b9be252631b2fbfc0683d4c1128/pom.xml#L276

Here is the link to my forks javadoc
https://helltux.github.io/freedomotic/

Command:
mvn clean javadoc:javadoc scm-publish:publish-scm

**Comment by mcicolella** [comment]
I agree with you about the second way. I think the javadoc should be updated only by project's mantainers or better automatically when our CI system creates a new dailybuild.
@Helltux Is it possible to push to a subfolder of gh-pages e.g. /javadoc and not to the root?
@bestmazzo What do you think?

**Comment by binhonglee** [comment]
Is this issue still open? And is there a reason why #133 was not merged?

**Comment by P3trur0** [comment]
@mcicolella what do you think? IMHO #229 Java code seems good, although I have no clue if the added Maven Plugin is suitable for our needs.

@binhonglee are you familiar with that?
